### ISSUE TITLE
Cbackup "cbackup-1.1.2 -1.el7.noarch.rpm" Error (#8)

### ISSUE THREAD

**Issue by pv081079** [cBackup/main#53]
The initial page to select language work but when I do next to:

https://surancid01.ritta.local/cbackup/install/index.php?r=install%2Frequirements

This error appears:

Error (#8)
An internal server error occurred.

The above error occurred while the Web server was processing your request.

Please contact us if you think this is a server error. Thank you.

**Comment by Zayats93** [comment]
mysql and php version?

**Comment by jigeshthampan** [comment]
# mysql -V
mysql  Ver 8.0.21 for Linux on x86_64 (Source distribution)

# php -v
PHP 7.4.12 (cli) (built: Oct 27 2020 15:01:52) ( NTS )
Copyright (c) The PHP Group
Zend Engine v3.4.0, Copyright (c) Zend Technologies
    with Zend OPcache v7.4.12, Copyright (c), by Zend Technologies

**Comment by Zayats93** [comment]
You have two options:
1) use mysql 5 and php 7.0
2) after installing cbackup edit yii2 files to workaround error
You will encounter 2 errors. First, error with PHP string to open ro stream with rw permission. 2nd, change in mysql, which change system mysql fields name to uppercase.
You can enable debug in yii to see in which files you get the error.

**Comment by jigeshthampan** [comment]
Yes. I have resorted to mysql 5.7 and php 7.2 which resolved the issues.
Thanks

**Comment by ATX-250** [comment]
Maybe it is help to anybody. I just install cBackup with mariadb 10.5.13 and php-8. To fix this error your need do only one change in file cbackup/helpers/SystemHelper.php at 38 line. Change this:
$desc   = [0 => ["pipe", "r"], 1 => ["pipe", "w"], 2 => ["pipe", "w"]];
to this:
$desc   = [0 => ["pipe", "w"], 1 => ["pipe", "w"], 2 => ["pipe", "w"]];
Change "r" to "w".

**Comment by bungee-** [comment]
Can confirm the fix from @ATX-250 .
Running on Ubuntu 20.04, PHP 7.4.3, MySQL Ver 8.0.30-0ubuntu0.20.04.2 for Linux on x86_64
